### PR TITLE
Book#index and Book#show implementation

### PR DESCRIPTION
**Pull request by marcegl91**
### Summary

- Added Book#index and Book#show actions, also defined factories for user, rent and book and fixed naming in models to follow API requirements.

### Trello Card

https://trello.com/c/eKCgYnNv/21-crear-rutas-index-y-show-para-book

### CODE REVIEW

File: wbooks-api/app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      include Wor::Paginate
+
+      before_action :authenticate_user!
+
+      def index
+        @books = Book.all
```
Comment (facucompe) [inline wbooks-api/app/controllers/api/v1/books_controller.rb]: There is no need to use instance variables here

File: wbooks-api/app/controllers/api/v1/books_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module Api
+  module V1
+    class BooksController < ApplicationController
+      include Wor::Paginate
+
+      before_action :authenticate_user!
```
Comment (facucompe) [inline wbooks-api/app/controllers/api/v1/books_controller.rb]: We usually create an `ApiController` that is in charge or this. All the other controllers inheritates from this. You can keep it this way

File: wbooks-api/app/helpers/books_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module BooksHelper
```
Comment (facucompe) [inline wbooks-api/app/helpers/books_helper.rb]: Remove this empty module

File: wbooks-api/app/models/image.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+class Image < ApplicationRecord
+  has_one :book, dependent: :destroy
```
Comment (facucompe) [inline wbooks-api/app/models/image.rb:2]: I would use `dependent: :nullify` instead

File: wbooks-api/config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,11 @@
 Rails.application.routes.draw do
   mount_devise_token_auth_for 'User', at: 'auth'
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  namespace :api do
+    namespace :v1 do
+      get '/books', to: 'books#index'
```
Comment (facucompe) [inline wbooks-api/config/routes.rb]: `  resources :books, only: [:show, :index]`

**Review by facucompe — APPROVED** [review]

**Review by facucompe — CHANGES_REQUESTED** [review]
Sorry, i clicked the wrong  button

File: wbooks-api/app/controllers/api/v1/books_controller.rb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class BooksController < ApiController
+      def index
+        books = Book.all
+        render_paginated books, each_serializer: BookIndexSerializer
```
Comment (facucompe) [inline wbooks-api/app/controllers/api/v1/books_controller.rb]: you can do `render_paginated Book.all`

File: wbooks-api/app/controllers/api/v1/books_controller.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      end
+
+      def show
+        book = Book.find(params[:id])
+        render json: book, serializer: BookShowSerializer
```
Comment (facucompe) [inline wbooks-api/app/controllers/api/v1/books_controller.rb]: `render json: Book.find(params[:id])`

**Review by facucompe — DISMISSED** [review]

File: wbooks-api/app/controllers/api/v1/books_controller.rb (hunk 6 above)
Comment (floresv) [reply wbooks-api/app/controllers/api/v1/books_controller.rb]: I agree

File: wbooks-api/app/controllers/api/v1/books_controller.rb (hunk 7 above)
Comment (floresv) [reply wbooks-api/app/controllers/api/v1/books_controller.rb]: With facu rlz :metal:

File: wbooks-api/app/models/rent.rb [hunk 8]
```diff
@@ -1,6 +1,10 @@
 class Rent < ApplicationRecord
-  validates :user, :book, :borrowed_at, :returned_at, presence: true
+  validates :user, :book, :rented_from, :rented_to, presence: true
 
   belongs_to :user
   belongs_to :book
+
+  def active?
+    returned_at.blank?
```
Comment (floresv) [inline wbooks-api/app/models/rent.rb]: Suspicious :thinking:

File: wbooks-api/app/models/book.rb [hunk 9]
```diff
@@ -1,5 +1,10 @@
 class Book < ApplicationRecord
-  validates :genre, :author, :image, :title, :editor, :year, presence: true
+  validates :genre, :author, :title, :publisher, :year, presence: true
 
   has_many :rents, dependent: :destroy
+  belongs_to :image
+
+  def active_rent
+    rents.find(&:active?)
```
Comment (floresv) [inline wbooks-api/app/models/book.rb]: I thinks It's better using a model scope.. :thinking:

**Review by facucompe — APPROVED** [review]

**Review by floresv — APPROVED** [review]

**Pull request merged**